src/libstd/path.rs

```diff
@@ -192,6 +192,10 @@ pub enum Prefix<'a> {
     /// Prefix `C:` for the given disk drive.
     #[stable(feature = "rust1", since = "1.0.0")]
     Disk(#[stable(feature = "rust1", since = "1.0.0")] u8),
+
+    /// Scheme `file:` used on Redox
+    #[stable(feature = "rust1", since = "1.0.0")]
+    Scheme(#[stable(feature = "rust1", since = "1.0.0")] &'a OsStr),
```

Comment:
Adding this variant is a breaking change

Reply:
Understood, how should this be addressed?

Reply:
Good question. The docs around `Prefix` imply that it's meant to be Windows-exclusive, so perhaps defining another type is the way to go? However, the only place where `Prefix` seems to be used is as part of `Component`, which is also a fully public enum and can't be extended with another kind of prefix.

It doesn't look like this part of the standard library was designed with this kind of extensibility in mind. Maybe in the next epoch the definition of `Prefix` could be changed (not sure if epochs allow these kinds of changes)?

Alternatively, you could define a `std::os::redox::path::PathExt` trait that's implemented by `Path` and allows access to the prefix. However, this does not allow `Path::components` to work like it should...

Reply:
> not sure if epochs allow these kinds of changes

I've also thought of that, but I don't think it would work (though I'd love to be corrected!). How would that work? I guess you would somehow change the version of the `Components` enum and `components()` function being used depending on the epoch. But what if code using it then tried to pass the `Components` to code in the older epoch? Though perhaps an unlikely use case for this particular struct, that probably shouldn't error complaining that the types are different, and couldn't work because the code it's passed to won't know how to deal with the new variant.